index.py

```diff
@@ -176,20 +186,22 @@ def __init_completed(self):
         return True
 
     def __init_monit_config(
-            self, enable=False, target_host="8.8.8.8", cycles=1):
+            self, enable=False, target_host="8.8.8.8", iface="", cycles=1):
         if enable is False:
             rm("-rf", "/etc/monit/conf.d/keepalive")
             service("monit", "restart")
             return
 
-        config = """check host targethost with address {target_host}
-    if failed icmp type echo
-        count 3 with timeout 20 seconds
-    then exec "/bin/bash -c '/sbin/cell_mgmt power_off force && /bin/sleep 5 && /sbin/reboot -i -f -d'"
+        ifacecmd = "" if iface == "" or iface is None \
+                   else "-I {}".format(iface)
+        config = """check program ping-test with path "/bin/ping {target_host} {ifacecmd} -c 3 -W 20"
+    if status != 0
+    then exec "/bin/bash -c '/usr/sbin/cell_mgmt power_off force && /bin/sleep 5 && /sbin/reboot -i -f -d'"
```

Comment:
`/sbin/reboot` is a symbolic link to `/bin/systemctl`.
use `which reboot` can see the default reboot command use `/usr/local/sbin/reboot`.